compiler/damlc/daml-stdlib-src/DA/Internal/LF.daml

```diff
@@ -243,6 +243,26 @@ data AnyContractKey = AnyContractKey
   , getanyContractKeyTemplateRep : TemplateTypeRep
   }
 
+#ifdef DAML_GENMAP
+-- We do not have a general Eq instance for Any since Any is not exposed
+-- to users directly and that instance will crash on some values but
+-- we can have total Eq instances for AnyTemplate, AnyChoice and AnyContractKey
+-- since those are guaranteed to be serializable.
+eqAny : Any -> Any -> Bool
+eqAny = primitive @"BEEqual"
+
+instance Eq AnyTemplate where
+  AnyTemplate a == AnyTemplate b = eqAny a b
```

Comment:
Why does `AnyTemplate` not contain any form of type representation in contrast to the other `Any*` types?

Reply:
Because there is no need for it. The reason why we have it in `AnyChoice` is because we want `toAnyChoice @T1 Archive` to be different from `toAnyChoice @T2 Archive`. For the template that type parameter is represented by the template itself.